### PR TITLE
Introduce tmp partition

### PR DESCRIPTION
**Pull request by Deedone**

### CODE REVIEW

**Review by xakep-amatop — COMMENTED** [review]
do you need to add some changes to zephyr aos app in order to work with fstab or it can work with both simultaneously with block device?

File: boards/rcar_spider.overlay [hunk 1]
```diff
@@ -13,3 +13,47 @@
...
+		flash_sim0: flash_sim@0 {
+			compatible = "soc-nv-flash";
+			reg = <0x00000000 DT_SIZE_M(32)>;
+			erase-block-size = <1024>;
+			write-block-size = <4>;
+			partitions {
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+			/*
```
Comment (oleksiimoisieiev) [inline boards/rcar_spider.overlay:48]: alignment

File: boards/rcar_salvator_xs_m3.overlay [hunk 2]
```diff
@@ -17,3 +17,47 @@
...
+		flash_sim0: flash_sim@0 {
+			compatible = "soc-nv-flash";
+			reg = <0x00000000 DT_SIZE_M(32)>;
+			erase-block-size = <1024>;
+			write-block-size = <4>;
+			partitions {
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+			/*
```
Comment (oleksiimoisieiev) [inline boards/rcar_salvator_xs_m3.overlay:52]: alignment

**Review by xakep-amatop — APPROVED** [review]
I've not found any major issues, so I approve this pr
`Acked-by: Mykola Kvach <mykola_kvach@epam.com>`

**Review by al1img — APPROVED** [review]

**Pull request merged**